### PR TITLE
Added DiscordJS Guide, Font Awesome, Changed MatrixBots

### PR DESCRIPTION
**Pull request by OGMatrix**
**Font Awesome on Main:**
ss1

**Font Awesome on viewing a Icon:**
ss2

**DiscordJS Guide on Page:**
ss1
ss2

**If you need more Images just say it!**

### CODE REVIEW

File: websites/D/DiscordJS Guide/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "DiscordJS Guide",
+    "author": {
+      "name": "OGMatrix",
+      "id": "657290880947978241"
+    },
+    "category": "other",
+    "logo": "https://i.imgur.com/Zt4i2l3.png",
+    "thumbnail": "https://i.imgur.com/ojmsp7f.png",
+    "version": "1.0.1",
```
Comment (ririxidev) [inline websites/D/DiscordJS Guide/dist/metadata.json:11]: Version on release should be ``1.0.0``

File: websites/D/DiscordJS Guide/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "color": "#20f856",
+    "tags": [
+      "ogmatrix",
```
Comment (ririxidev) [inline websites/D/DiscordJS Guide/dist/metadata.json:14]: ``ogmatrix``, delete this tag. This tag is not affiliated with this website, or is it?

File: websites/F/Font Awesome/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      "name": "OGMatrix",
+      "id": "657290880947978241"
+    },
+    "category": "other",
+    "logo": "https://i.imgur.com/SnQ9HYz.png",
+    "thumbnail": "https://i.imgur.com/GJFmAgg.png",
+    "version": "1.0.0",
+    "color": "#0285f7",
+    "tags": [
+      "ogmatrix",
```
Comment (ririxidev) [inline websites/F/Font Awesome/dist/metadata.json:14]: ``ogmatrix``, delete this tag. This tag is not affiliated with this website, or is it?

File: websites/F/Font Awesome/presence.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        pData.buttons = [
+          {
+            "label": "View Icon",
+            "url": document.URL
+          }
+        ];
+      } catch (e) {
+        pData.details = "Browsing Page:";
+        pData.state = "Icons";
+      }
```
Comment (ririxidev) [inline websites/F/Font Awesome/presence.ts:35]: Instead of catching, just check if ``icon`` variable is null or not.

**Review by ririxidev — CHANGES_REQUESTED** [review]
You should check out things that I requested to change.
P.S. Check your last pull request, because I responded to your comment, that was unnecessary. If you wanted to blame me that you updated your presence, you should check what you did first then blame. Thank you!

Thanks,
ririxi (Contact: mail@ririxi.dev, ririxi#2721)

File: websites/F/Font Awesome/presence.ts [hunk 5]
```diff
@@ -19,19 +19,19 @@ pPresence.on("UpdateData", async () => {
       myParam: string = urlParams.get('q');
 
     if (myParam == null) {
-      try {
-        const icon: string = document.querySelector("#page-top > div.view.flex.flex-column.min-vh-100.db-pr > div.flex-grow-1.flex-shrink-0.flex-basis-auto > div > div.ph6-l > div > section > header > div.flex.flex-column.flex-row-xl.items-center-xl.justify-between-xl.mb2.mb4-l > h1 > span").textContent;
+      const icon: any = document.querySelector("#page-top > div.view.flex.flex-column.min-vh-100.db-pr > div.flex-grow-1.flex-shrink-0.flex-basis-auto > div > div.ph6-l > div > section > header > div.flex.flex-column.flex-row-xl.items-center-xl.justify-between-xl.mb2.mb4-l > h1 > span");
```
Comment (ririxidev) [inline websites/F/Font Awesome/presence.ts:22]: no any type! Thank you!

**Review by ririxidev — CHANGES_REQUESTED** [review]
You should check out ONE thing that I requested to change.

Thanks,
ririxi (Contact: mail@ririxi.dev, ririxi#2721)

**Review by ririxidev — APPROVED** [review]
~~LGTM!~~
P.S. I removed ``Changed DiscordBot List`` thing from title since there isn't anything about it.

File: websites/D/DiscordJS Guide/presence.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  clientId: "819865300173324288"
+}),
+  pBrowsing = Math.floor(Date.now() / 1000);
+
+pPresence.on("UpdateData", async() => {
+  const pData: PresenceData = {
+    largeImageKey: "icon2"
+  };
+
+  pData.startTimestamp = pBrowsing;
```
Comment (ririxidev) [inline websites/D/DiscordJS Guide/presence.ts:11]: oh no wait, sorry for that but could you merge it together 🤣  to pData object

File: websites/F/Font Awesome/presence.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+}),
+  pBrowsing = Math.floor(Date.now() / 1000);
+
+pPresence.on("UpdateData", async () => {
+  const pData: PresenceData = {
+    largeImageKey: "icon2"
+  },
+    pPage = window.location.pathname;
+
+  pData.startTimestamp = pBrowsing;
```
Comment (ririxidev) [inline websites/F/Font Awesome/presence.ts:12]: oh no wait, sorry for that but could you merge it together 🤣  to pData object.

**Review by ririxidev — CHANGES_REQUESTED** [review]
oops, sorry for that could you fix these things too 🤣 

Thanks,
ririxi (Contact: mail@ririxi.dev, ririxi#2721)

File: websites/F/Font Awesome/presence.ts (hunk 7 above)
Comment (OGMatrix) [reply websites/F/Font Awesome/presence.ts:12]: Idk what you mean with that?

Comment (OGMatrix) [reply websites/F/Font Awesome/presence.ts:12]: Please tell me what you mean with that @ririxidev

**Comment by OGMatrix** [comment]
I requested you to review it because I hope you answer my question

File: websites/F/Font Awesome/presence.ts (hunk 7 above)
Comment (SlowLife1661) [reply websites/F/Font Awesome/presence.ts:12]:
```suggestion
    largeImageKey: "icon2",
    startTimestamp: pBrowsing
  },
    pPage = window.location.pathname;
```

Comment (ririxidev) [reply websites/F/Font Awesome/presence.ts:12]: SlowLife already answered.

Comment (OGMatrix) [reply websites/F/Font Awesome/presence.ts]: > SlowLife already answered.

fixed

**Review by ririxidev — COMMENTED** [review]
So, because I see you modified DiscordBotList presence, so where is this bug fix then? Since I only see change that I requsted to move startTimestamp to object?

**Comment by OGMatrix** [comment]
> So, because I see you modified DiscordBotList presence, so where is this bug fix then? Since I only see change that I requsted to move startTimestamp to object?

I added Settings to deactivate/activate the Submit Page and or to deactivate/activate Buttons

**Comment by ririxidev** [comment]
Hi @OGMatrix,

Unfournetely, I don't see these changes in your commits! You should recommit them to your fork and branch that you do pull request from.

Thanks,
ririxi (Contact: mail@ririxi.dev, ririxi#2721)

**Comment by OGMatrix** [comment]
> Hi @OGMatrix,
> 
> Unfournetely, I don't see these changes in your commits! You should recommit them to your fork and branch that you do pull request from.
> 
> Thanks,
> ririxi (Contact: [mail@ririxi.dev](mailto:mail@ririxi.dev), ririxi#2721)

I did it.

**Comment by ririxidev** [comment]
Heya again, @OGMatrix!

Ah this github! God damn it! I still see only change about moved ``startTimestamp`` to presenceData object, but I don't see any change related to enabling/disabling buttons or any kind of added settings! If it is possible could you send me links to these commits where you did those changes?

Thanks,
ririxi (Contact: mail@ririxi.dev, ririxi#2721)

**Comment by OGMatrix** [comment]
> Heya again, @OGMatrix!
> 
> Ah this github! God damn it! I still see only change about moved `startTimestamp` to presenceData object, but I don't see any change related to enabling/disabling buttons or any kind of added settings! If it is possible could you send me links to these commits where you did those changes?
> 
> Thanks,
> ririxi (Contact: [mail@ririxi.dev](mailto:mail@ririxi.dev), ririxi#2721)

I can send you some Screenshots of the Code

**Comment by ririxidev** [comment]
I'm back with informations to you, @OGMatrix!

I asked one of our Presence Reviewers to check if there is some changes about ``DiscordBot List`` presence, but unfortunately, they said same thing like I said before - that there is nothing more than change that moves startTimestamp to object. So you should really recommit these changes with settings!
P.S. You should bump version for ``DiscordBot List`` presence.

Thanks,
ririxi (Contact: mail@ririxi.dev, ririxi#2721)

File: websites/D/Discord BotList/presence.ts [hunk 8]
```diff
@@ -53,6 +54,13 @@ botPresence.on("UpdateData", async () => {
         }
       ];
     } else if (botPage.includes("/bots/")) {
+      if (botPage.includes("/new")) {
+        if (sSubmit) {
+          botData.details = "Adding Bot";
+        } else {
+          return;
+        }
```
Comment (ririxidev) [inline websites/D/Discord BotList/presence.ts:62]: Could I know why you return it instead of just doing nothing on 'else'?

**Review by ririxidev — CHANGES_REQUESTED** [review]
Heya,
You should check thing that I requested to change:
+ You need to __BUMP__ version on ``DiscordBot List`` presence.

Thanks,
ririxi (Contact: mail@ririxi.dev, ririxi#2721)

**Comment by OGMatrix** [comment]
> Heya,
> You should check thing that I requested to change:
> 
> * You need to **BUMP** version on `DiscordBot List` presence.
> 
> Thanks,
> ririxi (Contact: [mail@ririxi.dev](mailto:mail@ririxi.dev), ririxi#2721)

I DID, how often I have to tell you that, its on Version 1.0.1 before I changed nth!

**Comment by SlowLife1661** [comment]
> > Heya,
> > You should check thing that I requested to change:
> > 
> > * You need to **BUMP** version on `DiscordBot List` presence.
> > 
> > Thanks,
> > ririxi (Contact: [mail@ririxi.dev](mailto:mail@ririxi.dev), ririxi#2721)
> 
> I DID, how often I have to tell you that, its on Version 1.0.1 before I changed nth!

SeMiD prettified and refactored your Presence that's why the version was bumped but you still need to bump it because you made a changes to it.

File: websites/D/Discord BotList/dist/metadata.json [hunk 9]
```diff
@@ -24,5 +24,17 @@
...
+      "id": "buttons",
+      "title": "Show Buttons",
+      "value": true
+    },    
+    {
+      "id": "submitP",
+      "title": "Show Submit Page",
+      "value": true
+    }
+  ]
```
Comment (Bas950) [inline websites/D/Discord BotList/dist/metadata.json:39]: settings are not correct and is missing values

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/D/Discord BotList/dist/metadata.json (hunk 9 above)
Comment (OGMatrix) [reply websites/D/Discord BotList/dist/metadata.json]: @Bas950 fixxed

**Review by ririxidev — APPROVED** [review]
LGTM!

**Review by Bas950 — CHANGES_REQUESTED** [review]
upload proof of the settings in the extension, and 2 screens of it in discord, one with setting enabled one with it disabled.

**Comment by OGMatrix** [comment]
> upload proof of the settings in the extension, and 2 screens of it in discord, one with setting enabled one with it disabled.

ss7
ss8
ss10

**Review by Bas950 — APPROVED** [review]

**Review by ririxidev — APPROVED** [review]
lgtm

**Pull request merged**